### PR TITLE
Check for exception in https->sendRequest()

### PR DESCRIPTION
**Pull request by pintomax**
Check for exception in https->sendRequest()

### CODE REVIEW

**Comment by pintomax** [comment]
@monika-Dianomic added generic catchall and check for empty URL

File: slack.cpp [hunk 1]
```diff
@@ -63,35 +63,74 @@ SimpleHttps	*https;
 	pair<string, string> header = make_pair("Content-type", "application/json");
 	headers.push_back(header);
 
+	if (m_url.empty())
```
Comment (MonikaSharma06) [inline slack.cpp:66]: Main issue has been resolved now but I think, here you should also trim it first. If I put `space` in the field, then in syslog-

`Sep  3 16:44:12 foglamp-ThinkPad-E450 FogLAMP Notifications[16987]: ERROR: Exception while sending notification to slack webhook   : basic_string::substr: __pos (which is 2) > this->size() (which is 1)`

Comment (pintomax) [reply slack.cpp:66]: Let's create another JIRA task for this and close this

**Review by MonikaSharma06 — APPROVED** [review]
m_url should be checked for empty after trim. Otherwise it results in confusing error logs.

**Comment by pintomax** [comment]
FOGL-3253 just created for URL check

**Pull request merged**